lib/wsdl.js

```diff
@@ -1206,7 +1206,98 @@ WSDL.prototype.toXML = function() {
+            return this.definitions.schemas[nsName].types[splitPrefix_Name.name];
+        } else if (splitPrefix_Name.name in this.definitions.schemas[nsName].complexTypes) {
+            return this.definitions.schemas[nsName].complexTypes[splitPrefix_Name.name];
+        } else
+            return undefined;
+    } else
+        return splitPrefix_Name.prefix + ':' + splitPrefix_Name.name;
+};
+
+WSDL.prototype.findxsiTypeSchema = function (obj) {
+    // Pick up the schema for the type specified in element's xsi:type attribute.
+    var xsiTypeSchema;
+    if (obj[this.options.attributesKey]) {
+        var xsiType = obj[this.options.attributesKey]['xsi:type'];
+        if (xsiType) {
+            var type = splitQName(xsiType);
+            var typeURI;
+            if (type.prefix === TNS_PREFIX) {
+                // In case of xsi:type = "MyType"
+                typeURI = this.xmlns[type.prefix] || this.xmlns.xmlns;
+            } else {
+                typeURI = this.xmlns[type.prefix];
+            }
+            var typeDef = this.findSchemaObject(typeURI, type.name);
+            if (typeDef) {
+                xsiTypeSchema = typeDef.description(this.definitions);
+            }
+        }
+    }
+    return xsiTypeSchema;
+};
+
+WSDL.prototype.findSchema = function (topSchema, name, originalName) {
+    var schema;
+    if (topSchema) {
+        // Find the current object's schema
+        if (topSchema[name + '[]'])
+            name = name + '[]';
+
+        if (topSchema.name) {
+            if (topSchema.name === 'complexType') {
+                schema = this.findSchema(topSchema.children[0], name, originalName);
+            } else if (topSchema.name === 'sequence') {
+                // Linear search among the elements of the sequence
+                for (var i = 0; i < topSchema.children.length; i++) {
+                    if (topSchema.children[i].$name === originalName) {
+                        schema = topSchema.children[i];
+                        break;
+                    }
+                }
+
+                if (schema && schema.name === 'element') {
+                    var splitPrefix_Name = splitQName(schema.$type);
+                    if (splitPrefix_Name.prefix !== 'xs') {
+                        schema = this.findInNamespace(splitPrefix_Name);
+                    } else {
+                        schema = schema.$type;
+                    }
+                }
+            } else if (topSchema.name === 'complexContent' && topSchema.children[0].name === 'restriction' && topSchema.children[0].$base === 'soapenc:Array') {
+                schema = topSchema.children[0].children[1];
+                if ('$ref' in schema && schema.$ref === 'soapenc:arrayType' && '$n1:arrayType' in schema) {
+                    schema = schema['$n1:arrayType'].substring(0, schema['$n1:arrayType'].length - 2);
+                    if (schema.substring(0, 2) !== 'xs') {
```

Comment:
I think using a RegExp instead of creating a substring and check for non-equal is more efficient here --> `/^xs/.test(schema)` should do the trick